AppServer/google/appengine/api/datastore_distributed.py

```diff
@@ -486,7 +486,7 @@ def _Dynamic_Next(self, next_request, query_result, request_id=None):
       del self.__queries[cursor_handle]
       return
 
-    if internal_cursor.get_offset() >= internal_cursor.get_count():
+    if query.limit() and internal_cursor.get_offset() >= query.limit():
```

Comment:
should we check `query.has_limit()` here? (and below)

Reply:
Yes. That's clearer. I wasn't considering the case when the client sets the limit to 0 (in which case, they should get 0 results instead of all of them).